eng/pipelines/common/templates/runtimes/run-test-job.yml

```diff
@@ -189,6 +194,13 @@ jobs:
 
     # Download and unzip managed test artifacts
     - ${{ if ne(parameters.corefxTests, true) }}:
+      - ${{ if eq(parameters.testBuildPhased, true) }}:
```

Comment:
I would assume we should always download the managed components here - in non-phased case, we would download all the managed tests, in the phased case, we'd download the <code>targetGeneric</code> tests and build the <code>targetSpecific</code> tests here. Once you manage to actually make the runtime pipeline clean, please double-check that the total number of tests reported by the infra matches other runtime pipeline runs without your change.

Reply:
I also cannot seem to find the place in the <code>run-test-job</code> template where we're compiling the <code>targetSpecific</code> tests in the phased mode. Please try to make sure we're not just dropping target-specific tests on the floor with your change.

Reply:
The `run-test-job` is never building tests.

In phased mode we download both builds and unzip both to the same directory to create an overlay.

Reply:
Hmm, can you please elaborate on where exactly the build of "targetSpecific" tests takes place? I was under the impression that "run-test-job" would be the natural place for this as otherwise we would need to spin up a full OS/arch matrix of additional jobs to build them and that incurs quite a bit of overhead w.r.t. allocating and provisioning the machines, checking out the GitHub repo and such.

Reply:
They are being built with the old `build-test` legs. Those legs build the native dependencies, the wrappers, the core_root ….  They just no longer have to build the generic tests.

Reply:
I see, so they do get built during "run-test-job", just not using a separate step. I'm quite fine with that.

Reply:
It's currently a separate step.  I think test build and run were separated to make it easier to rerun tests and possibly better categorize failures.

Reply:
If we want these to build during the `run-test-job`.  I can look at it.

Reply:
I'm fine with merging this change in without this modification. The main motivation for splitting build-test-job and run-test-job was performance and reliability - test compilation fails far less often than test executions so that it was painstaking to wait for another test build when retrying the failed execution. My expectation was that this would be amortized by the fact that there's just a small number of targetSpecific tests. I think you have a valid point that, at least as an initial reading, it's cleaner to keep this phase separate, at least we'll see how much time is spent building the target-specific tests. Based on perf we can then decide whether we want to modify the other pipelines or revert the pipeline change and instead move platform-specific test build to the run-test-job template.

Reply:
Since Mono is now using these tests (the merge conflict), once we are happy, we should modify Mono to use our approach too.